src/async.md

```diff
@@ -0,0 +1,17 @@
+# Async Rust
+
+"Async" is a concurrency model where multiple tasks are executed concurrently by
+executing each task until it would block, then switching to another task that is
+ready to make progress. The model scales to higher concurrency than threads
+because the per-task overhead is typically very low and operating systems
+provide means of efficiently selecting tasks that can make progress.
```

Comment:
nit: It sounds like there is inherent OS support for async/await. Perhaps reword to something like "the async/await implementation relies on OS primitive to efficiently select the next tasks to run".

Reply:
I think I addressed this, but I'm happy for more editing :)